passw0rd/src/main/kotlin/com/virgilsecurity/passw0rd/protocol/ProtocolContext.kt

```diff
@@ -70,24 +70,21 @@ class ProtocolContext private constructor(
                 clientSecretKey: String,
                 updateToken: String
         ): ProtocolContext {
-            if (appToken.isBlank()) Utils.shouldNotBeEmpty("appToken")
-            if (servicePublicKey.isBlank()) Utils.shouldNotBeEmpty("servicePublicKey")
-            if (clientSecretKey.isBlank()) Utils.shouldNotBeEmpty("clientSecretKey")
-
-            val (publicVersion, publicBytes) = Utils.parseVersionAndContent(
-                    servicePublicKey,
-                    Utils.PREFIX_PUBLIC_KEY,
-                    Utils.KEY_PUBLIC_KEY
+            requires(appToken.isNotBlank(), "appToken")
+            requires(servicePublicKey.isNotBlank(), "servicePublicKey")
+            requires(clientSecretKey.isNotBlank(), "clientSecretKey")
+
+            val (publicVersion, publicBytes) = servicePublicKey.parseVersionAndContent(
+                    PREFIX_PUBLIC_KEY,
+                    KEY_PUBLIC_KEY
             )
 
-            val (secretVersion, secretBytes) = Utils.parseVersionAndContent(
-                    clientSecretKey,
-                    Utils.PREFIX_SECRET_KEY,
-                    Utils.KEY_SECRET_KEY
+            val (secretVersion, secretBytes) = clientSecretKey.parseVersionAndContent(
+                    PREFIX_SECRET_KEY,
+                    KEY_SECRET_KEY
             )
 
-            if (publicVersion != secretVersion)
-                throw IllegalArgumentException("Public and Secret keys must have the same version.")
+            require(publicVersion == secretVersion) { "Public and Secret keys must have the same version." }
```

Comment:
Probably we will get ""Parameter 'Public and Secret keys must have the same version.' should not be empty" exception message, which is not correct.